data-reconciliation-app/src/services/authentication-service.ts

```diff
@@ -67,215 +26,37 @@ export interface IAuthenticationService {
-   * @param {string} userId userId to check if it exists
-   * @returns {ServiceResult<boolean>}
-   */
-  public isUser(userId: string): ServiceResult<boolean> {
     try {
-      if (!userId) {
-        return ServiceResult.Failed({
-          errorMessage: "Error: invalid user id",
-          errorType: "AuthenticationError",
-        });
-      }
-
-      const usersCerts = ccfapp.typedKv(
-        "public:ccf.gov.users.certs",
-        ccfapp.arrayBuffer,
-        ccfapp.arrayBuffer
-      );
-      const result = usersCerts.has(ccf.strToBuf(userId));
-      return ServiceResult.Succeeded(result);
-    } catch (ex) {
-      return ServiceResult.Failed({
-        errorMessage: "Error getting caller identity",
-        errorType: "AuthenticationError",
-        details: ex,
-      });
-    }
-  }
-
-  /**
-   * Checks if a member exists and active
-   * @see https://microsoft.github.io/CCF/main/audit/builtin_maps.html#members-info
-   * @param {string} memberId memberId to check if it exists and active
-   * @returns {ServiceResult<boolean>}
-   */
-  public isActiveMember(memberId: string): ServiceResult<boolean> {
-    try {
-      if (!memberId) {
-        return ServiceResult.Failed({
-          errorMessage: "Error: invalid member id",
-          errorType: "AuthenticationError",
-        });
-      }
-
-      const membersCerts = ccfapp.typedKv(
-        "public:ccf.gov.members.certs",
-        ccfapp.arrayBuffer,
-        ccfapp.arrayBuffer
-      );
-
-      const isMember = membersCerts.has(ccf.strToBuf(memberId));
-
-      const membersInfo = ccfapp.typedKv(
-        "public:ccf.gov.members.info",
-        ccfapp.arrayBuffer,
-        ccfapp.arrayBuffer
-      );
-
-      const memberInfoBuf = membersInfo.get(ccf.strToBuf(memberId));
-      const memberInfo = ccf.bufToJsonCompatible(memberInfoBuf) as CCFMember;
-      const isActiveMember = memberInfo && memberInfo.status === "Active";
+      const commonCaller = request.caller as unknown as ccfapp.AuthnIdentityCommon;
+      const authPolicy = this.validators.get(<CcfAuthenticationPolicyEnum>commonCaller.policy);
+      return authPolicy.validate(request);
 
-      return ServiceResult.Succeeded(isActiveMember && isMember);
     } catch (ex) {
-      return ServiceResult.Failed({
-        errorMessage: "Error: getting caller identity",
-        errorType: "AuthenticationError",
-        details: ex,
-      });
-    }
-  }
-
-  /**
-   * Check if caller's access token is valid
-   * @param {JwtAuthnIdentity} identity JwtAuthnIdentity object
-   * @returns {ServiceResult<boolean>}
-   */
-  public isValidJwtToken(identity: ccfapp.JwtAuthnIdentity): ServiceResult<boolean> {
-
-    if (!identity || !identity.jwt || !identity.jwt.payload || !identity.jwt.payload.sub) {
       return ServiceResult.Failed({
         errorMessage: "Error: invalid caller identity",
-        errorType: "AuthenticationError",
-      });
+        errorType: "AuthenticationError"
```

Comment:
Validators all return `ServiceResult<identityId>`. We don't need a `try/catch` here.

Reply:
One scenario that I have considered -  If a jwt token is created from a different Idp which we are not supporting in that case it will throw an exception in jwt-validation.validate(). 
Hence, I have a put a try/catch block in the auth service in order to hold any of these scenarios.